### ISSUE TITLE
Add support for roles in google_sql_user

### ISSUE THREAD

**Issue by enocom** [hashicorp/terraform-provider-google#10438]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment. If the issue is assigned to the "modular-magician" user, it is either in the process of being autogenerated, or is planned to be autogenerated soon. If the issue is assigned to a user, that user is claiming responsibility for the issue. If the issue is assigned to "hashibot", a community member has claimed the issue already.

### Description

The [SqlServerUserDetails](https://cloud.google.com/sql/docs/mysql/admin-api/rest/v1/users#SqlServerUserDetails) in the SQL Admin API supports specifying `serverRoles` that will be attached to a user on creation. This is a request to expose that functionality in the `google_sql_user` resource.

### New or Affected Resource(s)

* google_sql_user

### Potential Terraform Configuration

```tf
resource "google_sql_user" "users" {
  name     = "me@example.com"
  instance = google_sql_database_instance.master.name
  type     = "CLOUD_IAM_USER"
  roles     = [ "read", "write", "admin" ] # or whatever roles a user might have in a database
}
```

### References

* https://issuetracker.google.com/issues/195676939

**Comment by arnisoph** [comment]
Is there some documentation for `serverRoles`? Is this even PostgreSQL-specific? I haven't found it in the API docs nor in the [gcloud CLI docs](https://cloud.google.com/sdk/gcloud/reference/sql/users/create).

**Comment by enocom** [comment]
@arnisoph The best I've found is under the [SqlServerUserDetails](https://cloud.google.com/sql/docs/mysql/admin-api/rest/v1/users#SqlServerUserDetails), which in spite of the name, does seem to apply to all three engines. Other than that, a person would probably have to experiment with the API to be certain.

**Comment by Illidanek** [comment]
Yes please! This seems to be a basic feature that the provider should have.
Without this we are forced to set permissions via magic scripts 🤮

**Comment by villesau** [comment]
@enocom @rileykarson what about postgres & mysql?

**Comment by villesau** [comment]
I think there's fairly sizeable amount of postres & mysql databases managed by Google too. When could we expect the same feature to land in those as well? This is a feature that prevents the system from being fully managed by Terraform.

**Comment by enocom** [comment]
I agree this would be a useful addition to the Cloud SQL API so we could support all engines. 

Since this isn't a Google provider limitation, let's track this work here: https://issuetracker.google.com/issues/195676939. Meanwhile, I'll bring this to the team that handles the Admin API.

**Comment by melinath** [comment]
Reopening this ticket. It is not actually resolved for any sql database types, even sql server. Per https://github.com/GoogleCloudPlatform/magic-modules/pull/6165#pullrequestreview-1030611518 SqlServerUserDetails is actually read-only.

**Comment by roaks3** [comment]
Marking as a feature request because the only mention of this being available through other surfaces is the SqlServerUserDetails, which as mentioned above is read-only. If this can already be done through the API, someone please correct me and we can change this to a coverage gap.

**Comment by antdking** [comment]
The API now supports managing user roles for mysql and postgres

https://docs.cloud.google.com/sql/docs/postgres/add-manage-iam-users#add-db-roles

**Comment by flamein** [comment]
@enocom I guess the timing is now right to bring this to terraform, as all three engines now seem to support serverRoles?

**Comment by jraby** [comment]
https://github.com/hashicorp/terraform-provider-google/commit/e825fbaba8c5ab3cc5d97580f27fcd9bdddd8663

looks like it is already in

**Comment by flamein** [comment]
ah, okay, then we probably have to wait for the next release

**Comment by roaks3** [comment]
Yep, supported by a field called `database_roles`, added in https://github.com/GoogleCloudPlatform/magic-modules/pull/16202.

This will likely be released later today.

**Comment by flamein** [comment]
Now if only we had a way of adding roles from terraform, this would actually be useful 🤔

**Comment by flamein** [comment]
It also doesn't seem to have made it into the docs https://registry.terraform.io/providers/hashicorp/google/7.18.0/docs/resources/sql_user

**Comment by melinath** [comment]
@flamein thanks! Docs have been added in https://github.com/GoogleCloudPlatform/magic-modules/pull/16300, which will roll out with the next release.
